core/src/test/java/org/apache/iceberg/jdbc/TestJdbcUtil.java

```diff
@@ -18,14 +18,116 @@
  */
 package org.apache.iceberg.jdbc;
 
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
 import java.util.Map;
 import java.util.Properties;
 import org.apache.iceberg.relocated.com.google.common.collect.Maps;
 import org.assertj.core.api.Assertions;
 import org.junit.jupiter.api.Test;
+import org.sqlite.SQLiteDataSource;
 
 public class TestJdbcUtil {
 
+  @Test
+  public void testUpdate() throws Exception {
+    SQLiteDataSource dataSource = new SQLiteDataSource();
+    dataSource.setUrl("jdbc:sqlite:file::memory:?icebergDB");
```

Comment:
I think we should have a test where the "old" schema with data is prepared (like you do here) and then we'd init the catalog pointing to that uri and do list tables/views operations to check how the catalog behaves after an upgrade

Reply:
Good idea, I will prepare a new test in `TestJdbcCatalog`/`TestJdbcViewCatalog`.

Reply:
In addition to just checking whether the schema has been updated correctly, we need to also test catalog operations (listTables() / listViews() / ...).

We should also have a test where we have the old schema with some data and then start the catalog with default settings (`jdbc.add-view-support` is set to `false`) to make sure `catalog.listTables()` (and others) work as expected and return correct results.

Reply:
Yup, I'm adding these tests. Thanks !

Reply:
I added `jdbc.add-view-support` property.